tests/browsercontext-proxy.spec.ts

```diff
@@ -91,6 +91,56 @@ it('should use proxy', async ({ contextFactory, server, proxyServer }) => {
   await context.close();
 });
 
+it.describe.parallel('should proxy local network requests', () => {
+  for (const additionalBypass of [false, true]) {
+    it.describe.parallel(additionalBypass ? 'with other bypasses' : 'by default', () => {
+      for (const params of [
+        {
+          target: 'localhost',
+          description: 'localhost',
+        },
+        {
+          target: '127.0.0.1',
+          description: 'loopback address',
+        },
+        {
+          target: '169.254.3.4',
+          description: 'link-local'
+        }
+      ]) {
+        it(`${params.description}`, async ({ platform, browserName, contextFactory, server, proxyServer }) => {
+          it.fail(platform !== 'linux' && browserName === 'webkit' && additionalBypass && ['localhost', '127.0.0.1'].includes(params.target), 'WK fails to proxy 127.0.0.1 and localhost if additional bypasses are present');
```

Comment:
nit, style: given the complexity of the fail predicate, I'd copy-paste this test 3 times instead.

Reply:
> nit, style: given the complexity of the fail predicate, I'd copy-paste this test 3 times instead.

@pavelfeldman how strong of a nit is this? Hehe 

I found the more complex fail predicate much easier to read/maintain/edit than trying to find the subtle differences in the otherwise lots of setup in each of the cases. 

Mind if I revert https://github.com/microsoft/playwright/commit/d307a872d3a5a432f82fb88cc71c63c50b03e01c before merging?

Reply:
> how strong of a nit is this?

Not too strong, do it the way you like.

Reply:
Happy to merge as is if this is ready.

Reply:
Thanks @pavelfeldman! Just reverted! Ready for a merge whenever assuming tests pass.